### ISSUE TITLE
Question on batch normalization.

### ISSUE THREAD

**Issue by rafaelvalle** [zhangxiangxiao/Crepe#22]
Why did you not use batch normalization on the Crepe network described in Text Understanding from Scratch? I looked at the paper but couldn't find anything that mentions it...
Does it have to do with the sparsity of the input data?

Thanks a lot for your work!

**Comment by EdwardDixon** [comment]
Check the date on the paper - this work was being done back when batch
normalisation was being invented.

On 28 Feb 2017 6:12 pm, "Rafael Valle" <notifications@github.com> wrote:

Why did you not use batch normalization on the Crepe network described in
Text Understanding from Scratch? I looked at the paper but couldn't find
anything that mentions it...
Does it have to do with the sparsity of the input data?

Thanks a lot for your work!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/zhangxiangxiao/Crepe/issues/22>, or mute the thread

**Comment by rafaelvalle** [comment]
Oh, I didn't keep track of the dates. 
Nonetheless, I would like to know if the authors would use batch norm. 
I'm using the Crepe network as the critic for training a generator using adversarial loss. Interestingly, with batch norm the generator outputs trash, whereas without batch norm it is capable of producing decent output. Any thoughts on this?

**Comment by zhangxiangxiao** [comment]
This issue was discussed via email. I am closing this issue, with my answer pasted below.

> 1. At the time I wrote code for the project, batch normalization is not published.
>
> 2. I tried to use batch normalization after seeing the paper. It blows up the model because the input has variable length and 0 is padded afterwards. For the last several features in length dimension, the variance is almost zero and when backpropagated the gradients blew up.
